tools/src/launcher/Main.hs

```diff
@@ -517,11 +527,70 @@ spawnNode (path, args, mbLogPath) = do
             logInfo "Node has started"
             return (ph, asc)
 
-runWallet :: (FilePath, [Text]) -> M ExitCode
-runWallet (path, args) = do
-    logNotice "Starting the wallet"
-    view _1 <$>
-        liftIO (readProcessWithExitCode path (map toString args) mempty)
+runWallet :: Bool -> (FilePath, [Text], Maybe FilePath) -> M ExitCode
+runWallet shouldLog (path, args, mLogPath) = do
+    customLogger LInfo NWallet "Starting the wallet"
+    phvar <- newEmptyMVar
+    liftIO $ case mLogPath of
+        Just lp -> do
+            createDirectoryIfMissing True (directory lp)
+            (_, logHandle) <- (lp,) <$> openFile lp AppendMode
+            liftIO $ IO.hSetBuffering logHandle IO.LineBuffering
+            let cr = createProcHandle path args logHandle
+            system' phvar cr mempty
+        Nothing ->
+           if shouldLog then
+               liftIO $ do
+               let cr = createProcInherit path args
+               system' phvar cr mempty
+           else
+              liftIO $ do
+              (_, _, _, pid) <- runInteractiveProcess path (map toString args) Nothing Nothing
```

Comment:
Using `runInteractiveProcess` here looks really strange. It will create a pipe and the out of the spawned process will be sent to this pipe. Another end of this pipe is returned by this function (actually three handles for three standard streams) and discarded.
It makes more sense to me to use `CreateProcess` with `std_out` and `std_err` set to `NoStream`.

Reply:
I refactored that